msbuild/Xamarin.MacDev.Tasks/MsBuildTasks/Exec.cs

```diff
@@ -1,8 +1,61 @@
-﻿using Xamarin.MacDev.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Xamarin.Messaging.Build.Client;
+using System.Security;
+using System.Reactive.Linq;
 
 namespace Microsoft.Build.Tasks
 {
-	public class Exec : ExecBase
+	public class Exec : ExecBase, ITaskCallback
 	{
+		public string ServerPassword { get; set; }
+
+		public override bool Execute ()
+		{
+			if (string.IsNullOrEmpty (SessionId))
+				return base.Execute ();
+
+			if (string.IsNullOrEmpty (ServerPassword))
+				return new TaskRunner (SessionId, BuildEngine4).RunAsync (this).Result;
+
+			return RunSudoCommandAsync ().Result;
+		}
+
+		public bool ShouldCopyToBuildServer (ITaskItem item) => false;
+
+		public bool ShouldCreateOutputFile (ITaskItem item) => false;
+
+		public IEnumerable<ITaskItem> GetAdditionalItemsToBeCopied () => Enumerable.Empty<ITaskItem> ();
+
+		async Task<bool> RunSudoCommandAsync ()
+		{
+			try {
+				var client = BuildConnection
+					.GetAsync (SessionId, BuildEngine4)
+					.Result
+					.Client;
+				var sshCommands = client
+					.MessagingService
+					.Ssh
+					.Commands;
+				var commandOutput = await sshCommands
+					.RunSudoCommandAsync (Command, ServerPassword.ToSecureString ())
```

Comment:
Yes, that's right, but that's just for running the sudo command. We don't use this from our targets but it's there in case an user wants to execute the sudo command remotely in a custom target.